src/Package/Impl/Repl/Commands/SourceRScriptCommand.cs

```diff
@@ -42,8 +42,14 @@ public SourceRScriptCommand(IRInteractiveWorkflow interactiveWorkflow, IActiveWp
                 // Save file before sourcing
                 ITextView textView = GetActiveTextView();
                 if (textView != null) {
-                    textView.SaveFile();
-                    _interactiveWorkflow.Operations.SourceFile(filePath, _echo);
+                    if (RPackage.Current != null) {
+                        textView.SaveFile();
+                    }
+
+                    var encoding = textView.TextBuffer.GetTextDocument()?.Encoding;
+
+                    _interactiveWorkflow.ActiveWindow?.Container.Show(false);
```

Comment:
There is no reason to call `SourceFile` if `ActiveWindow` became null between `SetStatus` and `Handle` calls